File: 129_sum_root_to_leaf_numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int sum=0;
    public int sumNumbers(TreeNode root) {
        if(root==null) return 0;
        sumNumber(root,0);
        return sum;
    }

    private void sumNumber(TreeNode node, int number){
        number=number*10+node.val;
        if(node.left==null && node.right == null) sum += number;
        else{
            if(node.left!=null) sumNumber(node.left,number);
            if(node.right!=null) sumNumber(node.right,number);
        }
    }
}

File: python/129_sum_root_to_leaf_numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def __init__(self):
        self.res = 0

    def sumNumbers(self, root: TreeNode) -> int:
        self.dfs(root, "")
        return int(self.res)

    def dfs(self, node: TreeNode, accum: str):
        if not node:
            return

        accum += str(node.val)
        if not node.left and not node.right:
            self.res += int(accum)
            return

        self.dfs(node.left, accum)
        self.dfs(node.right, accum)

